cmd/main.go

```diff
@@ -355,3 +361,21 @@ func autoMigrationWithLeader(migrationLeader leader.ElectorInterface, db *gorm.D
 		return err
 	})
 }
+
+func uploadBaseISOWithLeader(uploadLeader leader.ElectorInterface, objectHandler s3wrapper.API, generator generator.ISOInstallConfigGenerator, log logrus.FieldLogger) error {
+	ctx := context.Background()
+	return uploadLeader.RunWithLeader(ctx, func() error {
+		exists, err := objectHandler.DoesObjectExist(ctx, s3wrapper.BaseObjectName)
```

Comment:
maybe we can check it even before entering waiting for leader? it will optimize it for all future service restarts

Reply:
OK